app/services/create_collection_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class CreateCollectionService
+  def call(params, account)
+    tag = params.delete(:tag)
+    account_ids = params.delete(:account_ids)
+    @collection = Collection.new(params)
+    @collection.account = account
+    @collection.local = true
+    @collection.tag = find_or_create_tag(tag)
```

Comment:
I think I would have passed it directly to the `new` call as something like:
```suggestion
    @collection = Collection.new(params.merge({ account:, local: true, tag: find_or_create_tag(tag) }))
```

but this works too